src/WebJobs.Script.Scaling/IWorkerTableLock.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Scaling
+{
+    /// <summary>
+    /// table lock handle
+    /// </summary>
+    public interface IWorkerTableLock
```

Comment:
This doesn't seem specific to a worker table - you might generalize by renaming it ILockHandle. E.g. if you end up adding another table, you'll use this same lock interface for it as well I presume